src/components/Products.js

```diff
@@ -1,48 +1,56 @@
 import React from 'react';
 import { Component } from 'react';
 import { connect } from 'react-redux';
-import { addOrderProductThunk } from '../store';
+import { addOrderProductThunk, updateProductThunk, setProductsThunk } from '../redux/store';
 
 class _Products extends Component {
   constructor() {
     super();
     this.addToCart = this.addToCart.bind(this);
+    this.updateInventory= this.updateInventory.bind(this);
   }
-  async addToCart(item) {
-    // console.log('item added to cart', item);
-    await this.props.addToCart(item);
+  componentDidMount(){
+    this.props.set
+  }
+  addToCart(item) {
+    console.log('item added to cart', item);
+    this.props.addToCart(item);
+  }
+  updateInventory(item){
+    console.log('item for update', item )
+    this.props.updateInventory({...item, inventory: item.inventory-1});
```

Comment:
This gets tricky, because if another user on another computer purchases twenty while we are busy shopping, this inventory number is a liar.